extensions/amp-story/1.0/amp-story.js

```diff
@@ -915,6 +903,19 @@ export class AmpStory extends AMP.BaseElement {
    * @private
    */
   layoutStory_() {
+    const needsDvhPolyfill =
+      !this.win.CSS?.supports?.('height: 1dvh') &&
+      !getStyle(this.win.document.documentElement, '--story-dvh');
+
+    const onResize = (size) => {
+      needsDvhPolyfill && this.polyfillDvh_(size);
+      this.onViewportResize_();
+    };
+
+    this.getViewport().onResize(onResize);
+
+    onResize(this.getViewport().getSize());
```

Comment:
This should not be needed in this PR?

Reply:
Upon further testing it doesn't work on initialization unless I move the call here -

Reply:
Yes please! This could have a lot of CLS implications, we need to look into it carefully cc @mszylkowski

Reply:
I have opened up #37520 and added you and @mszylkowski as a reviewer, I'll merge this PR after that one is in.